internal/storage/storage.go

```diff
@@ -86,6 +86,10 @@ func (s *Storage) NewWriteBatch() *WriteBatch {
 // NewCommitBatch creates a batch for commit operations.
 func (s *Storage) NewCommitBatch(cc CommitContext) (*CommitBatch, error) {
 	cb := newCommitBatch(s.db.NewBatch(), s.writeOpts)
+	if err := cb.batch.Set(commitContextKey, encodeCommitContext(cc, cb.cc), nil); err != nil {
+		_ = cb.Close()
+		return nil, err
+	}
 	if err := cb.batch.Set(encodeCommitContextKeyInternal(cc, cb.cck), nil, nil); err != nil {
```

Comment:
Is it for backwards compatibility?

Reply:
Yes, this is left to keep tests succeeding. I will do a follow-up PR to remove this after changing codes for recovery and synchronization of log streams.